hbase-thrift/src/main/java/org/apache/hadoop/hbase/thrift2/ThriftHBaseServiceHandler.java

```diff
@@ -160,7 +162,12 @@ public int hashCode() {
   public ThriftHBaseServiceHandler(final Configuration conf,
       final UserProvider userProvider) throws IOException {
     super(conf, userProvider);
+    long cacheTimeout = conf.getLong(HBASE_CLIENT_SCANNER_TIMEOUT_PERIOD,
```

Comment:
We should close the scanner when expiring it?

Reply:
Agree. Fixed in the latest commit.